extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -0,0 +1,254 @@
+    this.registerAction('toggle', this.toggle_.bind(this));
+    this.registerAction('open', this.open_.bind(this));
+    this.registerAction('close', this.close_.bind(this));
+  }
+
+  /**
+   * @private
+   */
+  adjustPadding_() {
+    const viewerPaddingTop = this.viewport_.getPaddingTop();
+    if (viewerPaddingTop) {
+      // viewerPaddingTop exists when AMP page is rendered inside search
+      // carousel.
+      const div = this.document_.createElement('div');
+      setStyles(div, {
+        'height': viewerPaddingTop + 'px',
+        'width': '100%',
+      });
+      const firstChild = this.element.firstChild;
+      this.element.insertBefore(div, firstChild);
+    }
+    if (this.isIosSafari_) {
+      //Compensate for IOS safari bottom navbar.
+      const div = this.document_.createElement('div');
+      setStyles(div, {
+        'height': IOS_SAFARI_BOTTOMBAR_HEIGHT,
+        'width': '100%',
+      });
+      this.element.appendChild(div);
+    }
+  }
+
+  /**
+   * Toggles the open/close state of the sidebar.
+   * @private
+   */
+  toggle_() {
+    if (this.documentElement_.classList.contains('amp-sidebar-open')) {
+      this.close_();
+    } else {
+      this.open_();
+    }
+  }
+
+  /**
+   * Reveals the sidebar.
+   * @private
+   */
+  open_() {
+    this.mutateElement(() => {
+      this.getViewport().addToFixedLayer(this.element);
+      this.createMask_();
+      this.viewport_.disableTouchZoom();
+      this.documentElement_.classList.add('amp-sidebar-open');
+      this.documentElement_.classList.remove('amp-sidebar-closed');
+      this.element./*REVIEW*/scrollTop = 1;
+    });
+  }
+
+  /**
+   * Hides the sidebar.
+   * @private
+   */
+  close_() {
+    this.mutateElement(() => {
+      this.documentElement_.classList.remove('amp-sidebar-open');
+      this.documentElement_.classList.add('amp-sidebar-closed');
+      this.getViewport().removeFromFixedLayer(this.element);
+      this.getViewport().restoreOriginalTouchZoom();
+    });
+  }
+
+  /**
+   * Checks if the sidebar only has the whitlisted custom amp- elements.
+   * @returns {boolean} True when only whitelited elements are present.
+   * @private
+   */
+  checkWhitelist_() {
+    const elements = this.element.getElementsByTagName('*');
+    let i = elements.length - 1;
+    while (i >= 0) {
+      const tagName = elements[i].tagName;
+      if (tagName.indexOf('AMP-') == 0) {
+        const isWhiteListed = assert(
+            WHITELIST_.indexOf(tagName) >= 0,
+            '%s can only contain the following custom tags: %s',
+            this.element, WHITELIST_);
+        if (!isWhiteListed) {
+          return false;
+        }
+      }
+      i--;
+    }
+    return true;
+  }
+
+  /**
+   * @private
+   */
+  createMask_() {
+    if (this.hasMask_) {
+      return;
+    }
+    const mask = this.document_.createElement('div');
+    mask.setAttribute('class', '-amp-sidebar-mask');
+    mask.addEventListener('click', () => {
+      this.toggle_();
```

Comment:
I think this should be `hide_` or such to avoid problems with double-click here. It's too easy to tap twice and too illogical to hide and show the menu :)